source/lac/petsc_precondition.cc

```diff
@@ -94,28 +81,39 @@ namespace PETScWrappers
     AssertThrow(ierr == 0, ExcPETScError(ierr));
   }
 
+  void
+  PreconditionBase::setup()
+  {
+    AssertThrow(pc != nullptr, StandardExceptions::ExcInvalidState());
 
+    PetscErrorCode ierr = PCSetUp(pc);
+    AssertThrow(ierr == 0, ExcPETScError(ierr));
+  }
```

Comment:
What is the end result of this function? A shell `pc` object that is not associated with any specific algorithm? Where is this function used and in what context is it useful? You mention in the documentation that this function need not usually be called, but it would be useful to know what contexts it is useful for.

Reply:
This is not only for the Shell but for all preconditioners. It allows calling the setup of the `PC` at an earlier stage.